girder/cli/build.py

```diff
@@ -63,6 +63,19 @@ def main(dev, mode, watch, watch_plugin, npm, reinstall):
         npmLockFile = os.path.join(staging, 'package-lock.json')
         if os.path.exists(npmLockFile):
             os.unlink(npmLockFile)
+
+        # Remove any lingering node_modules to ensure clean install
+        plugin_dirs = [path.replace('file:', '') for path in _collectPluginDependencies().values()]
```

Comment:
`_collectPluginDependencies()` returns a dict of `(package, version)`. Technically, the `file:...` string is a version specifier. Consider renaming the the `path` variable accordingly.

More importantly, the [`GirderPlugin.npmPackages`](https://github.com/girder/girder/blob/4d80f4af14e2476e951f52c3cf869b864d2b5e76/girder/plugin.py#L103) method, which populates this dict doesn't *have* to return a value prefixed by `file:`; this could be overridden by plugins to return something different. So, I'd filter to only use entries that are prefixed by `file:`.

Reply:
Wouldn't the filtering be redundant though, as if we run `path.replace` on a path without the `file:` it would just remain unchanged?

Reply:
Yes, but the later stages of the pipeline, particularly `shutil.rmtree` would all be run on something that's not a path (like `^1.4`).

Reply:
Gotcha, I've added a filter